src/creatureevent.cpp

```diff
@@ -62,7 +62,7 @@ bool CreatureEvents::registerEvent(Event_ptr event, const pugi::xml_node&)
 	}
 
 	// if not, register it normally
-	creatureEvents.emplace(creatureEvent->getName(), std::move(*creatureEvent));
```

Comment:
Maybe we don't need this, since queries will always be in lowercase 🤗

Reply:
Find will always be in lowercase, but if the event is logged in uppercase then you won't be able to find it.

Another option is to set the name to lowercase when creating the CreatureEvent, so getting the name with your getName method would always return the name in lowercase.